File: problems/53. Maximum Subarray/Solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        localMaxes = [0] * len(nums)
        previous = 0
        for i in range(0, len(nums)):
            localMaxes[i] = max(previous + nums[i], nums[i])
            previous = localMaxes[i]
        return max(localMaxes)

File: problems/53. Maximum Subarray/Solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        vector<int> localMaxes(nums.size());
        int previous = 0;
        for (size_t i = 0; i < nums.size(); i++) {
            localMaxes[i] = std::max(previous + nums[i], nums[i]);
            previous = localMaxes[i];
        }
        return *max_element(localMaxes.begin(), localMaxes.end());
    }
};